packages/flutter/lib/service_extensions.dart

```diff
@@ -0,0 +1,13 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Service extensions for the Flutter framework.
+///
+/// To use, import `package:flutter/service_extensions.dart`.
+///
+/// This library should mainly be used for debugging purposes or by Flutter
+/// tooling.
+library service_extensions;
```

Comment:
I don't think we need a new top-level library here. Sorry if my previous comment may have been misleading.

I think we can just encapsulate all the consts in an enum. That way, only the name of that enum enters the global namespace and they can be referred to with something like `ServiceExtensions.debugDumpApp`. I think now that enums have a `name` getter extension, they should work here as that allows you to get the string representation of the enum value that you need to register them in bindings.dart.

Reply:
done. Named WidgetsServiceExtensions because there will be one of these per library where service extensions are registered (e.g. rendering, foundation, etc.), so naming as ServiceExtensions would cause naming collisions.